### PR TITLE
[#8107][Platform] [API] Error handling in certificate controller.

### PR DESCRIPTION
**Pull request by mahendranbhat**
**Description:**

Use the new global exception handling in the Certificate controller.

**Test Plan:**

- Its mostly refactor. Unit tests are updated.
- Run server and test basic operations.
- In UI went through the flow where we are calling certificate API.
- Created new self signed certificate also downloaded some certs.
- The UI page is working fine ( GET API and LIST )

Sbt etst
```[[info] Test run finished: 0 failed, 0 ignored, 7 total, 0.231s
[info] Passed: Total 1303, Failed 0, Errors 0, Passed 1299, Skipped 4
[success] Total time: 1043 s (17:23), completed 12 May, 2021 10:54:22 PM

```

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java [hunk 1]
```diff
@@ -125,77 +110,62 @@ public Result getRootCert(UUID customerUUID, UUID rootCA) {
...
-    }
+    Form<CertificateParams> formData = formFactory.getFormDataOrBadRequest(
+        CertificateParams.class);
+    Customer.getOrBadRequest(customerUUID);
+    CertificateInfo certificate = CertificateInfo.getOrBadRequest(rootCA);
     if (!certificate.customerUUID.equals(customerUUID)) {
-      return ApiResponse.error(BAD_REQUEST, "Certificate doesn't belong to customer");
+      throw new YWServiceException(BAD_REQUEST, "Certificate doesn't belong to customer");
     }
     if (certificate.certType == CertificateInfo.Type.SelfSigned) {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java:156]: move these two if blocks to:
private Certificate.checkEditable();

and call inside setCustomCertInfo

File: managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java [hunk 2]
```diff
@@ -75,19 +70,13 @@ public Result upload(UUID customerUUID) {
       return ApiResponse.success(certUUID);
     } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java:71]: yeah, what if there are some logging present in catch block?

File: managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java [hunk 3]
```diff
@@ -151,109 +154,97 @@ public static UUID createRootCA(String nodePrefix, UUID customerUUID, String sto
...
+    }
+    LOG.info("Created Client CA for username {} signed by root CA {}.", username, rootCA);
+    return bodyJson;
   }
 
   public static UUID uploadRootCA(
     String label, UUID customerUUID, String storagePath,
     String certContent, String keyContent, Date certStart,
     Date certExpiry, CertificateInfo.Type certType,
-    CertificateParams.CustomCertInfo customCertInfo) throws IOException {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java:256]: Also increase the scope of try (If required to cover whole method so even IOException gets caught and throw YWSE(INTERNAL_ERROR)  (Seee below comment)

File: managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java [hunk 4]
```diff
@@ -151,109 +154,97 @@ public static UUID createRootCA(String nodePrefix, UUID customerUUID, String sto
   }
 
   public static JsonNode createClientCertificate(UUID rootCA, String storagePath, String username,
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java:156]: If all these are ultimatly going to result in YWSE(INTERNAL_ERROR) then just catch these exception s in this method and throw throw YWSE(INTERNAL_ERROR)
This avoids poluting method signitures with checked exceptions (Assuming that all of these are getting  treated in a same manner - as 500)

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java (hunk 4 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java]: Yeah, make sense.

File: managed/src/main/java/com/yugabyte/yw/models/CertificateInfo.java [hunk 5]
```diff
@@ -205,4 +236,29 @@ public ArrayNode getUniverseDetails() {
...
+    if (!certificate.getInUse()) {
+      if (certificate.delete()) {
+        LOG.info("Successfully deleted the certificate:" + certUUID);
+      }
+      else {
+        throw new YWServiceException(INTERNAL_SERVER_ERROR, "Unable to delete the Certificate");
+      }
+    }
+    else {
+      throw new YWServiceException(BAD_REQUEST, "The certificate is in use.");
```
[the remark this answers is not in the document]
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/models/CertificateInfo.java:251]: @sb-yb

[the remark this answers is not in the document]
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/models/CertificateInfo.java:251]: @sb-yb

**Review by sb-yb — APPROVED** [review]

**Pull request merged**